Mono.Addins/Mono.Addins.Database/Util.cs

```
@@ -220,15 +220,22 @@ public static string GetGacPath (string fullName)
 					foreach (var dir in Directory.GetDirectories (asmDir, "v*_" + versionDirName)) {
 						var dirName = Path.GetFileName (dir);
 						i = dirName.IndexOf ('_');
-						Version av;
-						if (Version.TryParse (dirName.Substring (1, i - 1), out av)) {
+
+						try
+						{
+							Version av = new Version(dirName.Substring(1, i - 1));
```

dirName.Substring could theorically fail silently here, whereas it could not before.